plugins/repoclosure.py

```diff
@@ -86,9 +99,12 @@ def _get_unresolved(self, arch=None):
         if self.opts.check:
             checkpkgs = set()
             for repo in self.opts.check:
-                for pkgs_filtered in available.filter(reponame=repo):
+                for pkgs_filtered in self.base.sack.query().available().filter(reponame=repo):
                     checkpkgs.add(pkgs_filtered)
-            pkgs.intersection_update(checkpkgs)
+            if self.opts.pkglist:
+                pkgs.intersection_update(checkpkgs)
```

Comment:
If I use '--pkg <pkg>' with '--check <repo>' i expect that package from repo is checked. But this expression check all pkgs from repo + pkg described in --pkg option.